sklearn/linear_model/_coordinate_descent.py

```diff
@@ -1634,6 +1635,14 @@ def fit(self, X, y, sample_weight=None):
 
         alphas = self.alphas
         n_l1_ratio = len(l1_ratios)
+
+        check_scalar_alpha = partial(
+            check_scalar,
+            target_type=numbers.Real,
+            min_val=0.0,
+            include_boundaries="neither",
```

Comment:
Not including 0 technically breaks backward compatibility. For example, the following works on `main` with warnings about setting regularization to 0:

```python
from sklearn.linear_model import LassoCV
from sklearn.datasets import make_regression
X, y = make_regression(noise=4, random_state=0)

# Works but raises a warning
reg = LassoCV(cv=5, random_state=0, alphas=[0, 1]).fit(X, y)
```

To be on the safe side and not break backward compatibly:

```suggestion
            include_boundaries="left",
```